CMakeLists.txt

```diff
@@ -235,6 +231,29 @@ if(CUDA_FOUND)
       message(WARNING "On some Unix systems CUDA 10.0+ requires CMake 3.12.2+; you use CMake ${CMAKE_VERSION}")
   endif()
 
+  # We want to compile as many targets as possible but different CUDA versions support different targets.
+  # Let's instead enable options based on what cuda version we have.
+  if((CUDA_VERSION VERSION_EQUAL "9.0" OR CUDA_VERSION VERSION_GREATER "9.0") AND CUDA_VERSION VERSION_LESS "11.0")
+    option(COMPILE_CUDA_SM35 "Compile GPU version with SM35 support" ON)
```

Comment:
This creates different sets of options, which are not known from the beginning. I can see potential issues here. It might be better to only *set values* of those options here instead of creating them; see the very first comment.

Reply:
See my first comment. How do you deal with overriding user input? Doesn't this get even more confusing?